### PR TITLE
Add FAQ section to site

### PR DESCRIPTION
**Pull request by ceedubs**
This is just a start, and there are some more items that we will
probably want to add.

The import section probably is really a start to #912, and may end up
simply referencing a separate import document, as discussed in #936. But
I thought it would be nice to get something out there as an interim
resolution.

### CODE REVIEW

File: docs/src/main/tut/faq.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// cats adds a toXor method to the standard library's Either
+val e: Either[String, Int] = Right(3)
+e.toXor
+
+// cats adds an orEmpty method to the standard library's Option
+val o: Option[String] = None
+o.orEmpty
+```
+
+**Note**: if you import `cats.implicits._` (the preferred method), you should _not_ also use imports like `cats.std.syntax._` or `cats.std.either._`. This can result in ambiguous implicit values that cause bewildering compile errors.
```
Comment (peterneyens) [inline docs/src/main/tut/faq.md]: I suspect you meant something different than `cats.std.syntax._` ?

Comment (ceedubs) [inline docs/src/main/tut/faq.md]: Oops quite right thanks. I wish I could have doobie verify inline code! I'll fix this but I might not have time today.

Comment (ceedubs) [inline docs/src/main/tut/faq.md]: Fixed.

**Comment by non** [comment]
This is a great start. Let's merge it and iterate! 👍

**Comment by ceedubs** [comment]
Invoking the Documentation Rule and merging :)

**Pull request merged**